courses/models.py

```diff
@@ -382,11 +382,17 @@ def get_audit_class(cls):
     def to_dict(self):
         return serialize_model_object(self)
 
-    def reactivate_and_save(self):
-        """Sets an enrollment to be active again and saves it"""
+    def deactivate_and_save(self, change_status, no_user=False):
+        """Sets an enrollment to inactive, sets the status, and saves"""
+        self.active = False
+        self.change_status = change_status
+        return self.save_and_log(None if no_user else self.user)
+
+    def reactivate_and_save(self, no_user=False):
```

Comment:
This method doesn't seem to be used, but maybe it should be in `create_<program/run>_enrollment`, where `Enrollment.objects.create(` could be changed to `get_or_create`, and if not created then reactivate.  This might deal with edge cases like the one I ran into.

Reply:
It's used in `ecommerce/api.py`. I just moved the existing method and added a kwarg. I'll wait to hear back from Peter about the desired behavior